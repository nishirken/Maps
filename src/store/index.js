export testInitialState from './test-initial-state';
export testServerInitialState from './test-server-initial-state';
export store from './create-store';
export testStore from './test-store';
