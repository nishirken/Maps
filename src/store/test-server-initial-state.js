import { fromJS } from 'immutable';

export default fromJS([
    {
        index: 1,
        coords: {
            lng: 100,
            lat: 200,
            x: 20,
            y: 21,
        },
        name: 'Marker 1',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
    {
        index: 2,
        coords: {
            lng: 200,
            lat: 300,
            x: 10,
            y: 11,
        },
        name: 'Marker 2',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
]);
