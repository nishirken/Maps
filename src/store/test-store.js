import configureStore from 'redux-mock-store';
import initialState from './test-initial-state';

const mockStore = configureStore();
const store = mockStore(initialState.toObject());

export default store;
