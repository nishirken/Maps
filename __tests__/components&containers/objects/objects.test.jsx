import store from 'Store';
import { Objects } from 'Components';

describe('Objects component', () => {
    const ObjectsShallow = shallow(<Objects store={store} />);

    it('Rendered', () => {
        expect(ObjectsShallow).toMatchSnapshot();
    });
});
