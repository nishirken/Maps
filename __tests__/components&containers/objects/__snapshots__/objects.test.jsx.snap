exports[`Objects component Rendered 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    open={undefined}>
    <styled.div
        onClick={[Function]}>
        create new object
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div
      open={undefined}>
      <styled.div
            onClick={[Function]}>
            create new object
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Objects
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Objects {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "objectName": "",
        "props": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": Object {
          "createObject": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          open={undefined}>
          <styled.div
                    onClick={[Function]}>
                    create new object
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          open={undefined}>
          <styled.div
                    onClick={[Function]}>
                    create new object
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Objects
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    } />,
}
`;
